src/dxvk/imgui/dxvk_imgui.cpp

```diff
@@ -1238,39 +1301,46 @@ namespace dxvk {
       ImGui::SetCursorPosX(x + startX + (thumbnailSize - extent.x) / 2.f);
       ImGui::SetCursorPosY(y + (thumbnailSize - extent.y) / 2.f);
 
-      if (ImGui::ImageButton(pair.second.texID, extent)) {
-        const char* action;
-        if (data.find(pair.first) != data.end()) {
-          data.erase(pair.first);
-          action = "removed";
-        }
-        else {
-          data.insert(pair.first);
-          action = "added";
-        }
-
-        char buffer[256];
-        sprintf_s(buffer, "%s - %s %016llX\n", uniqueId, action, pair.first);
-        Logger::info(buffer);
+      if (ImGui::ImageButton(pair.second.texID, extent) && isListFiltered) {
+        toggleTextureSelection(pair.first, uniqueId, listRtxOption.textureSetOption->getValue());
       }
 
       if (ImGui::IsItemHovered()) {
         std::stringstream formatName;
         formatName << imageInfo.format;
 
+        //list all selections for this texture
+        std::string rtxTextureSelection;
+        for (auto& rtxOption : rtxTextureOptions) {
+          rtxOption.bufferTextureHash = pair.first;
+          rtxOption.bufferToggle = rtxOption.textureSetOption->getValue().find(pair.first) != rtxOption.textureSetOption->getValue().end();
+          if (rtxOption.bufferToggle) {
+            if (rtxTextureSelection.empty())
+              rtxTextureSelection = "\n";
+
+            rtxTextureSelection = str::format(rtxTextureSelection, rtxOption.displayName, "\n");
+          }
+        }
+
         char tooltip[1024];
-        sprintf(tooltip, "%s: %dx%d %s\nHash: 0x%" PRIx64,
+        sprintf(tooltip, "%s: %dx%d %s\nHash: 0x%" PRIx64 "\n%s\n\nRigth click to manage selection \n%s",
                 (imageInfo.usage & VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT) ? "Render Target" : "Texture",
                 imageInfo.extent.width, imageInfo.extent.height,
-                formatName.str().c_str() + strlen("VK_FORMAT_"), pair.first);
+                formatName.str().c_str() + strlen("VK_FORMAT_"), pair.first,
+                isLegacyMaterial ? "Legacy Material" : "Replaced Material",
+                rtxTextureSelection.c_str());
 
         ImGui::SetTooltip(tooltip);
-        if (ImGui::IsMouseDown(ImGuiMouseButton_Middle)) {
+        if (ImGui::IsMouseReleased(ImGuiMouseButton_Middle)) {
           ImGui::LogToClipboard();
           ImGui::LogText("%" PRIx64, pair.first);
           ImGui::LogFinish();
         }
 
+        if (ImGui::IsMouseReleased(ImGuiMouseButton_Right)) {
```

Comment:
After testing, I think Left click is more natural and inline with the previous texture interactions. Also works with Keyboard navigation.

Reply:
I've moved the context menu opening to be the texture button's action, so that it works with keyboard navigation and also doesn't reopen when clicking in another texture while the menu is already open